### ISSUE TITLE
Multiple same ids in generated DOM

### ISSUE THREAD

**Issue by rodriced** [perak/kitchen-site#148]
I have some dataview components in my application and It's very difficult to access to one specific element in the DOM because of multiple same ids. When I use $('#id-of-dom-element') it's seems that it's always the first elelement found which is returned.
All ids should be unique...
This constraint is explained in HTML W3C recommandation : http://www.w3.org/TR/html5/dom.html#the-id-attribute

**Comment by perak** [comment]
:+1:

**Comment by perak** [comment]
What do you suggest, how to generate ids (they cannot be random - because you need to access them from your code). Maybe "dataview-1", "dataview-2" ... ?

**Comment by rodriced** [comment]
@perak Yes you could add an auto generated suffix like an auto increment to ensure the uniqueness of ids, but you could add  a parameter id_suffix in component (only ?) when the user need to access to a part of a component.
And you must also pay attention when you loop over an html structure -> ids incremental suffix

**Comment by kadirerdogan** [comment]
How about COMPONENT_NAME-dataview?

**Comment by rodriced** [comment]
Why not ? For now it seems (I don't know very much kitchen...) that a component isn't reusable. So I think its name should be unique.
But perhaps in the future we will be able to reuse a component in different places.

**Comment by kadirerdogan** [comment]
After my first comment I remembered that TEMPLATE_NAME in the template files are replaced by the name from the json file (Not directly name but a combination of names :). 

So if you modify your templates and put ```TEMPLATE_NAME-``` in front of all ids you should get uniq ids.

@perak correct me if i am wrong.

Btw what do you mean by "component isnt reusable". Components are reusable but I guess you are talking about generated components. 

It would be nice to use components like we can use queries. Create an application level component and place it to multiple pages via its name. @perak it's a feature request :)

**Comment by kadirerdogan** [comment]
To make a warning don't modify template files directly since re-installation of the meteor-kitchen will remove the modified templates. 

There are two solution to this
* You can clone a template folder and modify cloned one (i use this mostly)
* You can copy the specific template file to a folder and use custom_component (perak's suggestion :)

**Comment by perak** [comment]
@kadirerdogan I'm experiencing flu here... thanks for answering!

@rodriced ids will be unique in v0.9.53, I didn't uploaded it yet, I hope tomorrow aprox 24 hours from now).

**Comment by rodriced** [comment]
@perak :+1:  Get well soon !

**Comment by perak** [comment]
@rodriced thank you :) Feeling better today... preparing 0.9.53...

**Comment by perak** [comment]
Not finished 0.9.53 yet, but this is how IDs work now (in forms and dataviews)

Unique ID is added to top-level html element (and `<form>` tag now doesn't have ID at all).

Here is snippet from generated code:

```
<template name="CustomersInsertForm">
  <div id="customers-insert-form">
    <h2 id="component-title">
      New customer
    </h2>
    <form role="form">
    ...
```

As you can see, top level html element got ID constructed from template name converted from CamelCase to kebab-case (`CustomersInsertForm` to `customers-insert-form`).

Sub-elements still have non-unique IDs (will be fixed in the future), but now you can access them with for example: `$('#customers-insert-form #component-title')...`

I hope this helps.

**Comment by rodriced** [comment]
Thanks very much for the reactivity !

**Comment by perak** [comment]
:+1:

v0.9.53 is uploaded just now. Enjoy! :)
